packages/agent/src/routes/modification/dissociate-delete-related.ts

```diff
@@ -28,14 +28,18 @@ export default class DissociateDeleteRelatedRoute extends RelationRoute {
   }
 
   public async handleDissociateDeleteRelatedRoute(context: Context): Promise<void> {
-    await this.services.authorization.assertCanDelete(context, this.collection.name);
-
     // Parse route params
     const parentId = IdUtils.unpackId(this.collection.schema, context.params.parentId);
     const isDeleteMode = Boolean(context.request.query?.delete);
     const caller = QueryStringParser.parseCaller(context);
     const filter = await this.getBaseForeignFilter(context);
 
+    if (isDeleteMode) {
+      await this.services.authorization.assertCanDelete(context, this.foreignCollection.name);
+    } else {
+      await this.services.authorization.assertCanEdit(context, this.collection.name);
```

Comment:
Hmmm, it should also be on the `foreignCollection`. 👀 

```suggestion
      await this.services.authorization.assertCanEdit(context, this.foreignCollection.name);
```